packages/aws-cdk/test/api/cloudformation-deployments.test.ts

```diff
@@ -142,3 +161,751 @@ test('if toolkit stack cannot be found but SSM parameter name is present deploym
 
   expect(requestedParameterName!).toEqual('/some/parameter');
 });
+
+test('readCurrentTemplateWithNestedStacks() can handle non-Resources in the template', async () => {
+  const cfnStack = instanceMockFrom((CloudFormationStack as any));
+  CloudFormationStack.lookup = jest.fn().mockImplementation((_, stackName: string) => {
```

Comment:
Is there any reason to use mocks here? Can't we just use a straight-up lambda function?

Same comment everywhere below when we use `jest.fn()`, but not do any assertions on the mock later.